### PR TITLE
Remove ZMQ version agnostic code

### LINKED ISSUE #67: Remove ZMQ version related code
I noticed quite a bit of code for handling old ZMQ versions. Since we bundle ZMQ 4.1.5 we could remove it.
Should we do this?

### PR DESCRIPTION
**Pull request by lgeiger**
Closes #67 

If we want to support linking libzmq installed on the system to support platforms other than linux, mac and windows, this should not be merged.

### CODE REVIEW

**Comment by lgeiger** [comment]
> If you think this is good to go and that it won't be hard to sync changes with upstream (in both directions), I say go for it.

That's the one thing which makes me hesitate. I don't know.

I had this branch on my machine for some time now and figured I just share it to discuss. We can definitely wait till it's clearer which route we want to go with `zmq-prebuilt`.

File: binding.cc [hunk 1]
```diff
@@ -1359,13 +1292,7 @@ namespace zmq {
     std::copy(dat, dat + len, cp);
     while (true) {
       int rc;
-    #if ZMQ_VERSION_MAJOR == 2
-      rc = zmq_send(socket->socket_, &msg, flags);
-    #elif ZMQ_VERSION_MAJOR == 3
-      rc = zmq_sendmsg(socket->socket_, &msg, flags);
```
Comment (lgeiger) [inline binding.cc]: Yes, look at line below this.

**Comment by minrk** [comment]
This seems fine to me. Plus, limiting support doesn't quite mean you are totally cutting off system-supplied zmq. It's just limiting the possible versions.

However, I would like to know what problems the presence of this code causes. If it doesn't directly cause real problems, I'm not sure what the benefit of removing it would be.

Personally, I'm trying hard to resist my own impulses to remove old code (it's super hard!) in projects people actually use until its presence causes an actual problem (impeding a refactor, relying on something else broken, etc.).

**Comment by lgeiger** [comment]
I feel the same. I made this PR mainly because I couldn't resist "my own impulses to remove old code" 😉.

I don't know of a problem caused by this code and the only benefit would be a more readable codebase and a minor improvement in test coverage. I think it'd make sense to close this PR and keep the branch around until we discover a problem caused by the code.

**Pull request closed without merging**